### ISSUE TITLE
Add birthday to the wallet database

### ISSUE THREAD

**Issue by kcalvinalvin** [bitcoindevkit/bdk_wallet#110]
**Describe the enhancement**  
Add a method `wallet_birthday` or `birthday` that returns the unix time of when the wallet was created.

**Use case**  
If we know the birthday of a wallet, we can avoid scanning blocks that happened before the wallet was created. If there's no birthday, we must scan the entire chain to be sure that we don't miss txs.

**Comment by luisschwab** [comment]
Claiming this one. What are everyone's thoughts on `timestamp` versus `height + hash`?  With `timestamp` the date could be set at wallet creation time assuming a `std` environment. With `height + hash`, we could only set that after connecting to a chain source and receiving a block.

**Comment by luisschwab** [comment]
@notmandatory I'll pick this one up, as rob closed 348. I've been busy with work on Floresta this past month.

**Comment by SIDHARTH20K4** [comment]
is anyone working on this issue currently?!

**Comment by luisschwab** [comment]
Yeah, this has been on my backlog for a while but I'm working on it right now.

**Comment by SIDHARTH20K4** [comment]
I'd be happy to review/test it when you're ready. Let me know if you want a second pair of eyes or need help testing edge cases.

**Comment by tnull** [comment]
Relatedly, it would also be cool if the wallet would still allow for some kind of 'reset'/override mode that still allows to force-rescan from a certain height, even if a certain birthday / local chain state is set. (cc @luisschwab).

**Comment by luisschwab** [comment]
> Relatedly, it would also be cool if the wallet would still allow for some kind of 'reset'/override mode that still allows to force-rescan from a certain height, even if a certain birthday / local chain state is set. (cc [@luisschwab](https://github.com/luisschwab)).

We could add an optional start height parameter to a sync request, which would override the birthday.
